server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3569,10 +3569,10 @@ public VlanVO doInTransaction(final TransactionStatus status) {
                     }
                     // increment resource count for dedicated public ip's
                     _resourceLimitMgr.incrementResourceCount(vlanOwner.getId(), ResourceType.public_ip, new Long(ips.size()));
-                } else if (domain != null) {
+                } else if (domain != null && !forSystemVms) {
                     // This VLAN is domain-wide, so create a DomainVlanMapVO entry
-                    //final DomainVlanMapVO domainVlanMapVO = new DomainVlanMapVO(domain.getId(), vlan.getId());
-                    //_domainVlanMapDao.persist(domainVlanMapVO);
+                    final DomainVlanMapVO domainVlanMapVO = new DomainVlanMapVO(domain.getId(), vlan.getId());
```

Comment:
the rest of the change seems trivial. I just wonder why these lines where commented in the first place. Can you explain @nvazquez ?

Reply:
as i can say. we tested it. it works now!

Reply:
@nvazquez is there any problem to merge it?